src/o365/graph/commands/teams/teams-messagingsettings-set.ts

```diff
@@ -0,0 +1,176 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import Utils from '../../../../Utils';
+import * as request from 'request-promise-native';
+import GraphCommand from '../../GraphCommand';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  allowUserEditMessages?: string;
+  allowUserDeleteMessages?: string;
+  allowOwnerDeleteMessages?: string;
+  allowTeamMentions?: string;
+  allowChannelMentions?: string;
+  teamId: string;
+}
+
+class GraphTeamsMessageSettingsSetCommand extends GraphCommand {
+  private static props: string[] = [
+    'allowUserEditMessages',
+    'allowUserDeleteMessages',
+    'allowOwnerDeleteMessages',
+    'allowTeamMentions',
+    'allowChannelMentions'
+  ];
+
+  public get name(): string {
+    return `${commands.TEAMS_MESSAGINGSETTINGS_SET}`;
+  }
+
+  public get description(): string {
+    return 'Updates messaging settings of a Microsoft Teams team';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    GraphTeamsMessageSettingsSetCommand.props.forEach(p => {
+      telemetryProps[p] = (args.options as any)[p];
```

Comment:
@rabwill one additional thing is that I reverted one unnecessary `if` statement that I added so instead of 


```
public getTelemetryProperties(args: CommandArgs): any {
    const telemetryProps: any = super.getTelemetryProperties(args);
    GraphTeamsMessageSettingsSetCommand.props.forEach((p: string) => {
      if (typeof (args.options as any)[p] !== 'undefined') {
        telemetryProps[p] = (args.options as any)[p].toLowerCase();
      }
    });
    return telemetryProps;
  }
```

it changed to 


```
public getTelemetryProperties(args: CommandArgs): any {
    const telemetryProps: any = super.getTelemetryProperties(args);
    GraphTeamsMessageSettingsSetCommand.props.forEach((p: string) => {
        telemetryProps[p] = (args.options as any)[p].toLowerCase();
    });
    return telemetryProps;
  }
```

Reply:
You are amazing! Thanks heaps for all you do @velingeorgiev

Reply:
Thanks!